app/controllers/order_items_controller.rb

```diff
@@ -0,0 +1,82 @@
+class OrderItemsController < ApplicationController
+  before_action :check_if_product_is_in_cart, only: [:create]
+
+  def create
+    @order = current_order
+    @order_item = @order.order_items.create(order_item_params)
+    if @order_item.save
+      redirect_to cart_path
+    else
+      if @order_item.product.inventory == 1
+        flash[:error] = "Sorry, there is only #{@order_item.product.inventory} #{@order_item.product.name} available."
+      elsif @order_item.product.inventory == 0
+        flash[:error] = "Sorry, there are no #{@order_item.product.name.pluralize} available."
+      else
+        flash[:error] = "Sorry, there are only #{@order_item.product.inventory} #{@order_item.product.name.pluralize} available."
+      end
+      redirect_to product_path(@order_item.product)
+    end
+  end
+
+  def destroy
+    @order = current_order
+    @order_item = @order.order_items.find(params[:id])
+    @order_item.destroy
+    @order_items = @order.order_items
+
+    redirect_to cart_path
+  end
+
+  def update
+    @order_item = OrderItem.find(params[:id])
+    @order_item.update_attributes(order_item_params)
+    if @order_item.save
+      redirect_to cart_path
+    else
+      flash[:error] = "Sorry, there are only #{@order_item.product.inventory} #{@order_item.product.name.pluralize} available."
+      redirect_to cart_path
+    end
+  end
+
+  def check_if_product_is_in_cart
+    order = current_order
+    @product_id = params[:order_item][:product_id]
```

Comment:
I don't think `product_id` needs to be a class variable here, as we're not using it outside of this method, or within any of our view code.